File: kt/src/main/kotlin/id/algo/Solution_268.kt
package id.algo

class Solution_268 {
    fun missingNumber(nums: IntArray): Int {
        val map = HashMap<Int, Int>(nums.size)
            .apply {
                for (i in 0..nums.size) {
                    this[i] = 0
                }
            }
            .apply {
                for (i in nums) {
                    this[i] = this.getOrDefault(i, 0) + 1
                }
            }
        println(map)
        for (entry in map.entries.iterator()) {
            if (entry.value == 0)  {
                return entry.key
            }
        }
        return -1
    }
}

File: sc/src/main/scala/Solution_268.scala
object Solution_268 {
  def missingNumber(nums: Array[Int]): Int = {
    var map = (0 to nums.size).map(e => (e, 0)).toMap
    for (i <- nums) {
      map = map.updated(i, map.getOrElse(i, 0) + 1)
    }
    for (entry <- map) {
      if (entry._2 == 0) {
        return entry._1
      }
    }
    -1
  }
}
